src/adera/Machines/Rocket.cpp

```diff
@@ -240,20 +285,43 @@ void SysMachineRocket::attach_plume_effect(ActiveEnt ent)
         return;
     }
 
-    m_scene.reg_emplace<ACompExhaustPlume>(plumeNode, ent, plumeEffect);
+    rScene.reg_emplace<ACompExhaustPlume>(plumeNode, mach, plumeEffect);
 }
 
-Machine& SysMachineRocket::instantiate(ActiveEnt ent, PrototypeMachine config,
-    BlueprintMachine settings)
+// TODO: come up with a better solution to this, and also replace config maps
+//       and variants with something else entirely
+template<typename TYPE_T>
+TYPE_T const& config_get_if(
+        std::map<std::string, config_node_t, std::less<>> const& nodeMap,
+        std::string_view field, TYPE_T&& defaultValue)
+{
+    auto found = nodeMap.find(field);
+    if (found == nodeMap.end())
+    {
+        return defaultValue;
+    }
+    return std::get<TYPE_T>(found->second);
+//    if (TYPE_T const* value = std::get_if<TYPE_T>(&found->second); value != nullptr)
+//    {
+//        return *value;
+//    }
+//    return defaultValue;
+}
+
+MachineRocket& SysMachineRocket::instantiate(
+        osp::active::ActiveScene& rScene,
+        osp::active::ActiveEnt ent,
+        osp::PrototypeMachine const& config,
+        osp::BlueprintMachine const& settings)
 {
     // Read engine config
     MachineRocket::Parameters params;
-    params.m_maxThrust = std::get<double>(config.m_config["thrust"]);
-    params.m_specImpulse = std::get<double>(config.m_config["Isp"]);
+    params.m_maxThrust = config_get_if<double>(config.m_config, "thrust", 42.0);
```

Comment:
The [] operators previously used here worked even though the values didn't exist. Changing stuff to const meant that at() had to be used instead, which caused errors.

Reply:
ah, it was creating the item if it didn't exist. Makes sense.

Also, note that for std::map<std::string, blah>, using operator[] with a char* allocates a new string in order to do the comparison, even if you're not creating a new item.